src/Middleware/HttpLogging/src/W3CLoggerOptions.cs

```diff
@@ -0,0 +1,117 @@
+            get { return _fileSizeLimit; }
+            set
+            {
+                if (value <= 0)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value), $"{nameof(FileSizeLimit)} must be positive.");
+                }
+                _fileSizeLimit = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets a strictly positive value representing the maximum retained file count or null for no limit.
+        /// Defaults to <c>4</c>.
+        /// </summary>
+        public int? RetainedFileCountLimit
+        {
+            get { return _retainedFileCountLimit; }
+            set
+            {
+                if (value <= 0)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value), $"{nameof(RetainedFileCountLimit)} must be positive.");
+                }
+                _retainedFileCountLimit = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets a string representing the prefix of the file name used to store the logging information.
+        /// The current date plus a file number (in the format {YYYYMMDD.X} will be appended to the given value.
+        /// Defaults to <c>w3clog-</c>.
+        /// </summary>
+        public string FileName
+        {
+            get { return _fileName; }
+            set
+            {
+                if (string.IsNullOrEmpty(value))
+                {
+                    throw new ArgumentNullException(nameof(value));
+                }
+                _fileName = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets a string representing the directory where the log file will be written to
+        /// Defaults to <c>/logs/</c> relative to the app directory (ContentRoot).
+        /// </summary>
+        public string LogDirectory
+        {
+            get { return _logDirectory; }
+            set
+            {
+                if (string.IsNullOrEmpty(value))
+                {
+                    throw new ArgumentNullException(nameof(value));
+                }
+                _logDirectory = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the period after which logs will be flushed to the store.
+        /// Defaults to 1 second.
+        /// </summary>
+        public TimeSpan FlushPeriod
+        {
+            get { return _flushPeriod; }
+            set
+            {
+                if (value <= TimeSpan.Zero)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value), $"{nameof(FlushPeriod)} must be positive.");
+                }
+                _flushPeriod = value;
+            }
+        }
+
+        /// <summary>
+        /// Fields to log. Defaults to logging request and response properties and headers.
+        /// </summary>
+        public W3CLoggingFields LoggingFields { get; set; } = W3CLoggingFields.Date | W3CLoggingFields.Time |
```

Comment:
@blowdart could you have a look at the fields being logged by default?

Reply:
I fixed up the request/request headers options to not include cookie